draft-zsf.md

```diff
@@ -0,0 +1,154 @@
+7. **Mitigate Payment-to-Self Vulnerabilities:** The risk of miners benefiting from "spoofing" transactions and the desire for a fixed eventual supply goal lead to considering alternatives like the Sustainability Fund, which pays out fees over a long enough time horizon to prevent recouping costs, allowing Zcash to remain closer to its supply goal and offering an alternative to other protocol designs that destroy or burn funds.
+8. **Recovery of "Soft-Burned" Funds:** In some instances, such as miners not claim all available rewards, some ZEC goes unaccounted for though not formally burned. This proposal would recover it through the `ZSF_BALANCE` mechanism described below.
+
+_Disclaimer 1: Long term success depends on the specific mechanisms of deposits, the quantities involved, and broader economic considerations. For such as if the payout rate is both sufficient and not excessive enough to threaten sustainability._
+
+# Specification
+
+In practice, The Zcash Sustainability Fund is a single global balance maintained by the node state and contributed to via a single transaction field. This provides the economic and security support described in the motivation section, while also importantly keeping the fund payouts extremely simple to describe and implement.
+
+The two modifications are:
+1. The re-accounting of unmined ZEC as a node state field called `ZSF_BALANCE`
+2. The addition of a transaction field called `ZSF_DEPOSIT`
+
+Please note that a **network upgrade is required** for this work to be fully implemented.
+
+## `ZSF_BALANCE`
+
+Upon activation at height `H`, the ZEC issuance mechanism is permanently suspended, the value of `ZSF_BALANCE[H]` is set to the remainder of unissued ZEC. This must be done before the `Block Rewards Payout Rule` is enforced for the block at height `H`. **This is not a pre-mine.** This is simply a re-accounting of unissued ZEC that comes with the benefit of other possible sources of funding.
+
+Consensus nodes are then required to track new per-block state:
+
+- `ZSF_BALANCE[H] : u64 [zatoshi]`
+
+The state is a single 64 bit integer (representing units of `zatoshi`) at any given block height, ``H``, representing the Sustainability Fund balance at that height, ``H``. The `ZSF_BALANCE` can be calculated using the following formula:
+
+`TOTAL ZEC TO EXIST (MAX_MONEY) - CLAIMED BLOCK SUBSIDIES OF PAST BLOCKS + SUM OF ALL ZSF DEPOSITS FROM PAST TRANSACTIONS`
+
+This formula holds for all future blocks. It is safe and semantically consistent to consider all blocks prior to the activation height to have a value of `0` in this field. This is not required by this proposal but may be convenient in designs or implementations.
+
```

Comment:
TODO for the next ZIP editors meeting:

Do we want to add a commitment to the `ZSF_BALANCE` node state as part of either:
- The existing block history commitment `hashChainHistoryRoot`: https://zips.z.cash/zip-0221
- An additional separate block commitment `hashSustainabilityFund`: https://zips.z.cash/zip-0244#block-header-changes

```draft suggestion
Each block commits to the previous block's `ZSF_BALANCE` in the block header by (TODO for ZIP editors)
```

Reply:
We looked through existing things that the block history commitment keeps track of, and it seems that we do not need to include it there. It would still be possible to use a FlyClient proof to prove that the sustainability fund balance has a particular value in a given block. The things that need to be explicitly included in the block history commitment are those that are useful to keep track of over a range, or that would improve light client security guarantees, which is not the case here.

Given that, it is more compatible to add this value to the end of the https://zips.z.cash/zip-0244#block-header-changes hash chain, because miners are required to be aware of extensions to this hash chain in any case. TODO: document guidelines for how to do that consistently.

Reply:
```suggestion
The block at height `H` commits to `ZsfBalanceAfter(H)` as a new `hashSustainabilityFund` in the block header, at the end of the `hashBlockCommitments` chain in [ZIP-244](https://zips.z.cash/zip-0244#block-header-changes).

TODO ZIP editors: consider introducing a chain state commitment hash tree. When we get closer to the network upgrade, we will have a better idea what commitments that network upgrade needs.
```